runhouse/resources/hardware/cluster.py

```diff
@@ -1595,6 +1599,116 @@ def _copy_certs_to_cluster(self):
         # private key should only live on the cluster
         Path(self.cert_config.key_path).unlink()
 
+    def run_bash(
+        self,
+        commands: Union[str, List[str]],
+        node: Optional[str] = None,
+        process: Optional[str] = None,
+        stream_logs: bool = True,
+        require_outputs: bool = True,
+    ):
+
+        if isinstance(commands, str):
+            commands = [commands]
+
+        if node is not None and process is not None:
+            raise ValueError("Only one of node or process can be specified.")
+
+        if node is None:
```

Comment:
question for this case
- this allows node and process to both not be specified, but wouldn't this throw an error when we call `run_bash_command_on_node_or_process`?
- noting that this differs from previous behavior, is this intended? before, if node is not specified, we set it to just the head ip, and to run on all nodes we supported explicitly setting `node="all"`